adapters/lora/src/main/java/org/eclipse/hono/adapter/lora/LoraCommandSubscriptionsHandler.java

```diff
@@ -0,0 +1,116 @@
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *******************************************************************************/
+
+package org.eclipse.hono.adapter.lora;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+
+import org.eclipse.hono.adapter.lora.providers.LoraProvider;
+import org.eclipse.hono.client.command.CommandConsumer;
+import org.eclipse.hono.notification.NotificationEventBusSupport;
+import org.eclipse.hono.notification.deviceregistry.LifecycleChange;
+import org.eclipse.hono.notification.deviceregistry.TenantChangeNotification;
+import org.eclipse.hono.tracing.TracingHelper;
+import org.eclipse.hono.util.TriTuple;
+
+import io.opentracing.Span;
+import io.opentracing.Tracer;
+import io.opentracing.noop.NoopTracerFactory;
+import io.opentracing.tag.Tags;
+import io.vertx.core.CompositeFuture;
+import io.vertx.core.Context;
+import io.vertx.core.Future;
+import io.vertx.core.Vertx;
+
+/**
+ * Class that holds the Lora adapter command subscriptions, that are shared between multiple {@link LoraProtocolAdapter}
+ * verticles. It also takes care to close the subscriptions for deleted/disabled tenants.
+ */
+public class LoraCommandSubscriptionsHandler implements LoraSubscriptionHandler {
+
+    private final Vertx vertx;
+    private final Tracer tracer;
+    private final Map<SubscriptionKey, TriTuple<CommandConsumer, LoraProvider, Context>> commandSubscriptions;
+
+    /**
+     * Creates a new Lora command subscription handler.
+     *
+     * @param vertx The Vert.x instance to use.
+     */
+    public LoraCommandSubscriptionsHandler(final Vertx vertx) {
+        this(vertx, null);
+    }
+
+    /**
+     * Creates a new Lora command subscription handler.
+     *
+     * @param vertx The Vert.x instance to use.
+     * @param tracer The tracer instance.
+     */
+    public LoraCommandSubscriptionsHandler(final Vertx vertx, final Tracer tracer) {
+        this.commandSubscriptions = new ConcurrentHashMap<>();
+        this.vertx = vertx;
+        this.tracer = tracer == null ? NoopTracerFactory.create() : tracer;
+    }
+
+    @Override
+    public void start() {
+        NotificationEventBusSupport.registerConsumer(vertx, TenantChangeNotification.TYPE,
+                notification -> {
+                    if (LifecycleChange.DELETE.equals(notification.getChange())
+                            || (LifecycleChange.UPDATE.equals(notification.getChange())
+                                    && !notification.isTenantEnabled())) {
+                        handleTenantTimeout(notification.getTenantId());
+                    }
+                });
+    }
+
+    private void handleTenantTimeout(final String tenantId) {
+        final Span span = TracingHelper
+                .buildSpan(tracer, null, "check command subscriptions on tenant timeout",
```

Comment:
Such a span here will be created every time a tenant gets deleted or disabled, regardless of whether there is a command subscriptions entry for that tenant.
It would be better to exit early and skip span creation here, if no corresponding command subscriptions map entries exist.